hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -589,23 +589,46 @@ public CompletableFuture<?> link(DataStream stream, LogEntryProto entry) {
     if (stream == null) {
       return JavaUtils.completeExceptionally(new IllegalStateException(
           "DataStream is null"));
+    } else if (!(stream instanceof LocalStream)) {
+      return JavaUtils.completeExceptionally(new IllegalStateException(
+          "Unexpected DataStream " + stream.getClass()));
     }
     final DataChannel dataChannel = stream.getDataChannel();
     if (dataChannel.isOpen()) {
       return JavaUtils.completeExceptionally(new IllegalStateException(
           "DataStream: " + stream + " is not closed properly"));
     }
 
-    final ContainerCommandRequestProto request;
     if (dataChannel instanceof KeyValueStreamDataChannel) {
```

Comment:
Let's change it to 
```java
    if (!(dataChannel instanceof KeyValueStreamDataChannel)) {
      return JavaUtils.completeExceptionally(new IllegalStateException(
          "Unexpected DataChannel " + dataChannel.getClass()));
    }
```
Then, the code don't need to have so many indentation levels.